### ISSUE TITLE
CythonGuile unit test fails with segfault on ubuntu 16.04 (also failing on jenkins)

### ISSUE THREAD

**Issue by mjsduncan** [opencog/atomspace#1532]
from `make test` output:
```
118/119 Test #118: CythonGuile ......................***Exception: SegFault  6.28 sec
test_a_load_core_types (test_pattern.SchemeTest) ... ok
test_b_load_file (test_pattern.SchemeTest) ... ok
test_c_eval (test_pattern.SchemeTest) ... ok
test_unifier (test_pattern.SchemeTest) ...
```
running individual test as per linus:
```
~/oc/atomspace$ export PYTHONPATH=./build/opencog/cython/
~/oc/atomspace$ nosetests -vs tests/cython/guile
test_a_load_core_types (test_pattern.SchemeTest) ... ok
test_b_load_file (test_pattern.SchemeTest) ... ;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /home/mjsd/oc/atomspace/./tests/cython/guile/basic_unify.scm
;;; compiled /home/mjsd/oc/guile/ccache/2.2-LE-8-3.A/home/mjsd/oc/atomspace/tests/cython/guile/basic_unify.scm.go
ok
test_c_eval (test_pattern.SchemeTest) ... ok
test_unifier (test_pattern.SchemeTest) ... ('\nThe question is:', (BindLink
  (VariableNode "$var") ; [6006671607988762398][1]
  (InheritanceLink
    (VariableNode "$var") ; [6006671607988762398][1]
    (ConceptNode "animal") ; [4997524230749066760][1]
  ) ; [10101611238486688056][1]
  (VariableNode "$var") ; [6006671607988762398][1]
) ; [9242815897039095064][1]
)
Segmentation fault (core dumped)
```

**Comment by linas** [comment]
stack trace? You can get one either with gdb, or there may be one in opencog.log in that directory.

Don't know what unifier this is testing -- perhaps this is a @ngeiswei  thing?

**Comment by mjsduncan** [comment]
the ubuntu error handling system apport now packages core dumps into .crash files and won't run dbg if the error wasn't generated by a package.  fishing out what i think is the core dump gives this:

dbg -c coredump
Fatal Python error: Py_Initialize: Unable to get the locale encoding
  File "/usr/lib/python2.7/encodings/__init__.py", line 123
    raise CodecRegistryError,\
                            ^
SyntaxError: invalid syntax

Current thread 0x00007fc1625a6700 (most recent call first):
Aborted (core dumped)

**Comment by DeborggraeveR** [comment]
Should be fixed by #1545

**Comment by linas** [comment]
closing I assume this is fixed.
